packages/ra-ui-materialui/src/list/filter/FilterForm.tsx

```diff
@@ -1,105 +1,41 @@
     ReactNode,
-    useEffect,
     useCallback,
     useContext,
+    useEffect,
 } from 'react';
-import { styled } from '@mui/material/styles';
-import {
-    FormGroupsProvider,
-    SourceContextProvider,
-    SourceContextValue,
-    useListContext,
-    useResourceContext,
-} from 'ra-core';
-import {
-    FieldValues,
-    FormProvider,
-    useForm,
-    useFormContext,
-} from 'react-hook-form';
-import set from 'lodash/set';
-import unset from 'lodash/unset';
-import get from 'lodash/get';
-import cloneDeep from 'lodash/cloneDeep';
-import isEqual from 'lodash/isEqual';
+import { useFormContext } from 'react-hook-form';
 
-import { FilterFormInput } from './FilterFormInput';
 import { FilterContext } from '../FilterContext';
+import { FilterFormInput } from './FilterFormInput';
 
 export const FilterForm = (props: FilterFormProps) => {
-    const { defaultValues, filters: filtersProps, ...rest } = props;
-
-    const { setFilters, displayedFilters, filterValues } = useListContext();
+    const { filters: filtersProps, ...rest } = props;
     const filters = useContext(FilterContext) || filtersProps;
 
-    const mergedInitialValuesWithDefaultValues =
-        mergeInitialValuesWithDefaultValues(
-            defaultValues || filterValues,
-            filters
-        );
-
-    const form = useForm({
-        defaultValues: mergedInitialValuesWithDefaultValues,
-    });
-    const { getValues, reset, trigger, watch } = form;
-
-    // Reapply filterValues when the URL changes or a user removes a filter
-    useEffect(() => {
-        const newValues = getFilterFormValues(getValues(), filterValues);
-        const previousValues = getValues();
-        if (!isEqual(newValues, previousValues)) {
-            reset(newValues);
-        }
-        // The reference to the filterValues object is not updated when it changes,
-        // so we must stringify it to compare it by value and also compare the reference.
-        // This makes it work for both input values and filters applied directly through
-        // the ListContext.setFilter (e.g. QuickFilter in the simple example)
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [JSON.stringify(filterValues), filterValues, getValues, reset]);
-
-    useEffect(() => {
-        const subscription = watch(async (values, { name }) => {
-            // We must check whether the form is valid as watch will not check that for us.
-            // We can't rely on form state as it might not be synchronized yet
-            const isFormValid = await trigger();
-
-            // Check that the name is present to avoid setting filters when watch was
-            // triggered by a change on the ListContext values.
-            if (name && isFormValid) {
-                if (get(values, name) === '') {
-                    const newValues = cloneDeep(values);
-                    unset(newValues, name);
-                    setFilters(newValues, displayedFilters, true);
```

Comment:
This piece of logic wasn't ported. Are you sure it wasn't required?

Reply:
I wasn't sure at first, but according to my tests, the new filterValues are properly cleaned up everytime. So apparently it's not required.
But maybe it would be safer to reintroduce it anyways. So I'll do it.